copylot/assemblies/photom/gui/windows.py

```diff
@@ -215,10 +218,83 @@ def initialize_UI(self):
         self.playButton.clicked.connect(self.play_drawing)
         self.playButton.hide()  # Initially hide the Play button
 
+        # dropdowns for drawing mode
+        self.drawing_dropdowns = QHBoxLayout()
+        self.drawing_dropdowns_widget = QWidget()
+
+        # roi dropdown box
+        self.roi_dropdown= QComboBox(self)
+        self.roi_dropdown.addItem("Select ROI")
+        self.photom_window.shapesUpdated.connect(self.updateRoiDropdown)
+        self.roi_dropdown.currentIndexChanged.connect(self.onRoiSelected)
+        self.drawing_dropdowns.addWidget(self.roi_dropdown)
+
+        # pattern dropdown box
+        self.patterns = ['Bidirectional']
+        self.pattern_dropdown = QComboBox(self)
+        self.pattern_dropdown.addItem("Select Pattern")
+        self.addPatternDropdownItems()
+        self.pattern_dropdown.currentIndexChanged.connect(self.onPatternSelected)
+        self.drawing_dropdowns.addWidget(self.pattern_dropdown)
+        self.drawing_dropdowns_widget.setLayout(self.drawing_dropdowns)
+
+        # horizontal spacing box for bidirectional pattern
+        self.spacing_boxes = QHBoxLayout()
+        self.horizontal_spacing_label = QLabel("Horizontal Spacing:", self)
+        self.spacing_boxes.addWidget(self.horizontal_spacing_label)
+        self.horizontal_spacing_input = QLineEdit(self)
+        self.spacing_boxes.addWidget(self.horizontal_spacing_input)
+
```

Comment:
```suggestion
```

Reply:
you can format this with black to avoid these extra spaces and inconsistent use of Quotes ("") vs ('')